torchaudio/compliance/kaldi.py

```diff
@@ -284,10 +287,10 @@ def spectrogram(waveform: Tensor,
         snip_edges, raw_energy, energy_floor, dither, remove_dc_offset, preemphasis_coefficient)
 
     # size (m, padded_window_size // 2 + 1, 2)
-    fft = torch.rfft(strided_input, 1, normalized=False, onesided=True)
+    fft = torch.rfft(strided_input, 1, normalized=False, onesided=True).to(dtype=dtype, device=device)
```

Comment:
This change does not seem to be necessary.
`strided_input` and `fft` should be already on the right dtype/device.
I tried your code without this change and tests ran fine.
Did you find a case this causes `RuntimeError`?

Reply:
Reverted this change,unittest passed.
Before I got runtime error at torch.max step , so safe side  I  changed device and dtype for inputs to max